byzcoin/tx_pipeline.go

```diff
@@ -124,6 +120,12 @@ func (s *defaultTxProcessor) CollectTx() ([]ClientTransaction, error) {
 	root := proto.(*CollectTxProtocol)
 	root.SkipchainID = s.scID
 	root.LatestID = latest.Hash
+	// When a block is processed, we prevent conodes to send us back transactions
+	// until the next collection.
+	if !isNotProcessingBlock {
```

Comment:
So, if isNotProcessingBlock ever gets stuck to be false, then (1) we'll never pick up any more txns from followers, and (2) they will never require a view-change, because they continue to hear from the leader.

Also: the double negative here implies that maybe this flag should be flipped and called isProcessingBlock. Would at least make this check easier to reason about.

Reply:
It's actually called `doCatchUp` in other places as the purpose of the function is first to check if a block is stored or currently in the block buffer, but this naming here was unclear so I changed it to that to make it a bit more obvious on the purpose.

It won't be stuck because the block buffer is cleaned if the block fails to be created. It's up to the leader to be consistent on that. Now if you're afraid that a leader could send heartbeats without creating blocks, it was already the case. Not saying we shouldn't do anything but looks like another issue to open.